src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs

```diff
@@ -192,6 +192,11 @@ public partial class NewAzureRmVmssConfigCommand : Microsoft.Azure.Commands.Reso
             ValueFromPipelineByPropertyName = true)]
         public string[] IdentityId { get; set; }
 
+        [Parameter(
+           Mandatory = false,
+           ValueFromPipelineByPropertyName = true)]
+        public bool UltraSSDEnabled { get; set; }
```

Comment:
@hyonholee if that user provides no value for this parameter, what is the default value? It seems like it would be better to make this a switch parameter called `-EnableUltraSSD` that the user can see if the default behavior is disabled, but I'm not sure if that works with the logic below (instantiating the objects sent to the server, etc.).

This comment applies for all instances of this new parameter.

Reply:
@hyonholee
Is there a distinction between the case when UltraSSDEnabled -s false vs Absent?

The code [here](https://github.com/Azure/azure-powershell/pull/7429/files#diff-b6f0ad3c65642019eab97e326814dc46R430) treats the cases differently and suggests that there is a difference b/w the setting not being specified vs the setting it to false. What is the difference in terms of the vm behavior/config int he two cases?

Reply:
@robhat1

Reply:
@praries880 If it’s a switch, then I don’t think we’ll be able to set it to false which is a legitimate scenario a customer might want to do with existing VM i.e. change it to false

Reply:
@singhkays what happens if we just set it to false when the switch is not present? is that different form the setting being present on the vm but being set to false?
i.e :
```
                vVirtualMachineProfile.AdditionalCapabilities.UltraSSDEnabled = this.UltraSSDEnabled;
```
This setting will be false when the switch is false and true when the switch is present, but its always applied to the vm. Is that not acceptable?

Reply:
This should be discussed before the Powershell design review was approved.  Anyway, I think the semantics for null value of AdditionalCapabilities and for false value of UItraSSDEnabled are different.   @robhat1 to confirm it.  (can we set it to false when the storage account type is not allowed for UltraSSD?)

Reply:
Talked to @rohbat1 offline. He'll respond here with the differences between false and null. Usually null in ARM APIs means nothing to change

Reply:
The property in CRP is a nullable Boolean. We don’t want it to be specified if user doesn’t want anything to do with UltraSSD. Sending a value of false by default would unnecessarily result in having CRP to persist the flag in its KVS.
Also, there is a difference between null and false.

This is the behavior that we want:
1.	Send a value true if UltraSSD has to be enabled.
2.	Send a value false if UltraSSD has to be disabled.
3.	Send null if user doesn’t have anything to do with UltraSSD or does not want any change to existing persisted value in CRP. If the persisted value in CRP is true and null is sent, CRP simply ignores it and assumes user does not want to change anything.

Reply:
@singhkays @robhat1 Thanks for the explanation...
So If im reading this right... the UltraSSD setting is for more than the current vm/vmss it is being applied to. The text above seems to suggest that the setting (if specified) is toggled for the user rather than just the vm or vmss being created here. 

Is my understanding correct? If it is then this seems like it should be an entirely different cmdlet itself that the user uses to toggle the UltraSSD setting n shouldn't be bundled with the new-AzVM(ss)config command.

Reply:
@hyonholee @robhat1 
I just synced with @singhkays offline.
For the  New-AzureRmVMConfiig, New-AzureRmVmssConfig c,dlets, i propose adding a switcvh called "EanbleUltraSSD", which if present sets the config to true. If the switch is not present, we do not  send in the configuration. Given that we are creating a new VM/Vmss, the effect is that ultraSSD is not enabled.

For Update-AuzreRmVM, and Update-AzureRmVmss i am fine with keeping it as a boolean value since here the three different states do make a difference.

Let me know if this sounds resonable.